src/io/flutter/run/bazelTest/FlutterBazelTestConfigurationEditorForm.java

```diff
@@ -12,42 +12,81 @@
   private JPanel myMainPanel;
+  private JLabel scopeLabelHint;
+  private JComboBox<Scope> scope;
 
   private JLabel myEntryFileLabel;
   private TextFieldWithBrowseButton myEntryFile;
+  private JLabel myEntryFileHintLabel;
 
-  private JLabel myLaunchingScriptLabel;
-  private TextFieldWithBrowseButton myLaunchingScript;
   private JTextField myBuildTarget;
+  private JLabel myBuildTargetHintLabel;
+  private JLabel myBuildTargetLabel;
+
+  private JLabel myTestNameLabel;
+  private JTextField myTestName;
+  private JLabel myTestNameHintLabel;
+
+  private Scope displayedScope;
 
   public FlutterBazelTestConfigurationEditorForm(final Project project) {
+    scope.setModel(new DefaultComboBoxModel<>(new Scope[]{TARGET_PATTERN, FILE, NAME}));
+    scope.addActionListener((ActionEvent e) -> {
+      final Scope next = getScope();
+      updateFields(next);
+      render(next);
+    });
+    scope.setRenderer(new ListCellRendererWrapper<Scope>() {
+      @Override
+      public void customize(final JList list,
+                            final Scope value,
+                            final int index,
+                            final boolean selected,
+                            final boolean hasFocus) {
+        setText(value.getDisplayName());
+      }
+    });
     final FileChooserDescriptor descriptor = FileChooserDescriptorFactory.createSingleFileDescriptor();
-    myLaunchingScript.addBrowseFolderListener("Select Launching Script", "Choose launching script", project, descriptor);
 
     DartCommandLineConfigurationEditorForm.initDartFileTextWithBrowse(project, myEntryFile);
   }
 
   @Override
   protected void resetEditorFrom(@NotNull final BazelTestConfig configuration) {
     final BazelTestFields fields = configuration.getFields();
+    myTestName.setText(fields.getTestName());
     myEntryFile.setText(FileUtil.toSystemDependentName(StringUtil.notNullize(fields.getEntryFile())));
     myBuildTarget.setText(StringUtil.notNullize(fields.getBazelTarget()));
-    myLaunchingScript.setText(FileUtil.toSystemDependentName(StringUtil.notNullize(fields.getLaunchingScript())));
+    final Scope next = fields.getScope();
+    scope.setSelectedItem(next);
+    render(next);
   }
 
   @Override
   protected void applyEditorTo(@NotNull final BazelTestConfig configuration) {
-    final BazelTestFields fields = new BazelTestFields();
-    fields.setEntryFile(StringUtil.nullize(FileUtil.toSystemIndependentName(myEntryFile.getText().trim()), true));
-    fields.setBazelTarget(StringUtil.nullize(myBuildTarget.getText().trim(), true));
-    fields.setLaunchingScript(StringUtil.nullize(FileUtil.toSystemIndependentName(myLaunchingScript.getText().trim()), true));
+    final String testName = StringUtil.nullize(this.myTestName.getText().trim(), true);
+    final String entryFile = StringUtil.nullize(FileUtil.toSystemIndependentName(myEntryFile.getText().trim()), true);
```

Comment:
Good call wrapping this with `FileUtil.toSystemIndependentName()`, when we have Windows behaving differently is is typically because I forgot to wrap with this method.